components/bt/host/bluedroid/stack/btm/btm_inq.c

```diff
@@ -2537,7 +2537,101 @@ void BTM_AddEirService( UINT32 *p_eir_uuid, UINT16 uuid16 )
 
 /*******************************************************************************
 **
-** Function         BTM_RemoveEirService
+** Function         btm_find_empty_custom_uuid_slot
+**
+** Description      Helper function for custom service managing routines.
+**
+** Parameters       custom_uuid - pointer to custom_uuid array in tBTA_DM_CB
+**                  uuid - UUID struct
+**
+** Returns          Slot number if there is empty slot,
+**                  otherwise - BTA_EIR_SERVER_NUM_CUSTOM_UUID
+**
+*******************************************************************************/
+static UINT8 btm_find_empty_custom_uuid_slot(tBT_UUID *custom_uuid, tBT_UUID uuid)
+{
+    for (UINT8 xx = 0; xx < BTA_EIR_SERVER_NUM_CUSTOM_UUID; xx++){
+        if (custom_uuid[xx].len == 0) {
+            return xx;
+        }
+    }
+    return BTA_EIR_SERVER_NUM_CUSTOM_UUID;
+}
+
+/*******************************************************************************
+**
+** Function         btm_find_match_custom_uuid_slot
+**
+** Description      Helper function for custom service managing routines.
+**
+** Parameters       custom_uuid - pointer to custom_uuid array in tBTA_DM_CB
+**                  uuid - UUID struct
+**
+** Returns          Slot number if given UUID is already in slots array,
+**                  otherwise - BTA_EIR_SERVER_NUM_CUSTOM_UUID
+**
+*******************************************************************************/
+static UINT8 btm_find_match_custom_uuid_slot(tBT_UUID *custom_uuid, tBT_UUID uuid)
+{
+    for (UINT8 xx = 0; xx < BTA_EIR_SERVER_NUM_CUSTOM_UUID; xx++) {
+        if (!memcmp (&custom_uuid[xx].uu, &uuid.uu, uuid.len)) {
+            return xx;
```

Comment:
Maybe there will be a mismatch between `custom_uuid[xx]`(`custom_uuid.len = LEN_UUID_128`) and `uuid`(`uuid.len = LEN_UUID_16`)? What do you think about matching tBT_UUID?

Reply:
Great catch!
I added a small `btm_compare_uuid()` that compares only UUIDs with similar lengths.